### ISSUE TITLE
Fix S2931 FP: Handle IAsyncDisposable as disposing interface

### ISSUE THREAD

**Issue by lg2de** [SonarSource/sonar-dotnet#3324]
### Description

When implementing "dispose" using `IAsyncDisposable` S2931 is complaining the class should implement `IDispose`.

### Repro steps

```cs
public class TestClass : IAsyncDisposable
{
    private CancellationTokenSource? cancellationTokenSource;

    public Task MethodAsync()
    {
        this.cancellationTokenSource = new CancellationTokenSource();
        return Task.Delay(1000, this.cancellationTokenSource.Token);
    }

    public void Stop()
    {
        this.cancellationTokenSource?.Cancel();
    }

    public ValueTask DisposeAsync()
    {
        this.cancellationTokenSource?.Dispose();
        return new ValueTask();
    }
}
```

### Expected behavior

S2931 should accept `IDisposable` and `IAsyncDisposable`.

### Actual behavior

S2931 only accepts `IDisposable`.

### Known workarounds

N/A

### Related information

* C# Plugins Version: 8.6.1.17183
* Visual Studio Version: 16.5.4
* MSBuild / dotnet version: dotnetcore 3.1
* Operating System: Windows 10
